app/assets/javascripts/Components/Modals/submission_file_upload_modal.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import Modal from 'react-modal';
+
+class SubmissionFileUploadModal extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      newfiles: []
+    }
+  }
+
+  componentDidMount() {
+    Modal.setAppElement('body');
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    this.props.onSubmit(this.state.newfiles);
+  };
+
+  handleFileUpload = (event) => {
+    this.setState({newfiles: event.target.files})
+  };
+
+  render() {
+    return (
+      <Modal
+        className="react-modal"
+        isOpen={this.props.isOpen}
+        onRequestClose={this.props.onRequestClose}
+      >
+        <h2>{I18n.t('add_new')}</h2>
```

Comment:
Oh man let's use this opportunity to get rid of this misnamed key. I'd be happy with `I18n.t('upload')` ("Upload"), or you can use `'upload_the'` and create a new generic key for `"File"` if you want.